In [8]:
# import the libraries

import pandas as pd

In [60]:
# import the datasets

adverse_df = pd.read_csv("adverse_reactions.csv")
patient_df = pd.read_csv("patients.csv")
treatment_df = pd.read_csv("treatments.csv")

### Assessment 

In [13]:
adverse_df.isnull().sum()

given_name          0
surname             0
adverse_reaction    0
dtype: int64

In [19]:
adverse_df.duplicated().sum()

0

In [21]:
adverse_df.sample(10)

given_name   surname   adverse_reaction
27     idalia     moore       hypoglycemia
29       alex  crawford       hypoglycemia
5     jasmine     sykes       hypoglycemia
10      tegan   johnson           headache
15     jia li      teng             nausea
2      joseph       day       hypoglycemia
31     steven       roy           headache
26    clinton    miller  throat irritation
30     monika    lončar       hypoglycemia
4     manouck   wubbels  throat irritation

In [23]:
adverse_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34 entries, 0 to 33
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   given_name        34 non-null     object
 1   surname           34 non-null     object
 2   adverse_reaction  34 non-null     object
dtypes: object(3)
memory usage: 944.0+ bytes


In [26]:
patient_df.isnull().sum()

patient_id       0
assigned_sex     0
given_name       0
surname          0
address         12
city            12
state           12
zip_code        12
country         12
contact         12
birthdate        0
weight           0
height           0
bmi              0
dtype: int64

In [28]:
patient_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 503 entries, 0 to 502
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   patient_id    503 non-null    int64  
 1   assigned_sex  503 non-null    object 
 2   given_name    503 non-null    object 
 3   surname       503 non-null    object 
 4   address       491 non-null    object 
 5   city          491 non-null    object 
 6   state         491 non-null    object 
 7   zip_code      491 non-null    float64
 8   country       491 non-null    object 
 9   contact       491 non-null    object 
 10  birthdate     503 non-null    object 
 11  weight        503 non-null    float64
 12  height        503 non-null    int64  
 13  bmi           503 non-null    float64
dtypes: float64(3), int64(2), object(9)
memory usage: 55.1+ KB


In [30]:
patient_df.head()

patient_id assigned_sex given_name  surname                    address  \
0           1       female        Zoe  Wellish       576 Brown Bear Drive   
1           2       female     Pamela     Hill  2370 University Hill Road   
2           3         male        Jae   Debord      1493 Poling Farm Road   
3           4         male       Liêm     Phan        2335 Webster Street   
4           5         male        Tim  Neudorf       1428 Turkey Pen Lane   

                city       state  zip_code        country  \
0  Rancho California  California   92390.0  United States   
1          Armstrong    Illinois   61812.0  United States   
2               York    Nebraska   68467.0  United States   
3         Woodbridge          NJ    7095.0  United States   
4             Dothan          AL   36303.0  United States   

                                      contact  birthdate  weight  height   bmi  
0        951-719-9170ZoeWellish@superrito.com  7/10/1976   121.7      66  19.6  
1       PamelaSHill@cuvox.de+1 (217) 569-3204   4/3/1967   118.8      66  19.2  
2            402-363-6804JaeMDebord@gustr.com  2/19/1980   177.8      71  24.8  
3  PhanBaLiem@jourrapide.com+1 (732) 636-8246  7/26/1951   220.9      70  31.7  
4             334-515-7487TimNeudorf@cuvox.de  2/18/1928   192.3      27  26.1

In [34]:
patient_df.duplicated().sum()

0

In [54]:
patient_df.given_name.value_counts()

John       9
Robert     3
Søren      3
Abel       2
Mustafa    2
          ..
Markus     1
Galit      1
Melisa     1
Peter      1
Pat        1
Name: given_name, Length: 470, dtype: int64

In [53]:
patient_df[patient_df.given_name == 'John']

patient_id assigned_sex given_name      surname                  address  \
140         141         male       John  Teichelmann    631 Isaacs Creek Road   
153         154         male       John     Carreiro  1463 Martha Ellen Drive   
215         216         male       John          Doe          123 Main Street   
229         230         male       John          Doe          123 Main Street   
237         238         male       John          Doe          123 Main Street   
242         243         male       John      O'Brian                      NaN   
244         245         male       John          Doe          123 Main Street   
251         252         male       John          Doe          123 Main Street   
277         278         male       John          Doe          123 Main Street   

         city state  zip_code        country  \
140   Decatur    IL   62522.0  United States   
153      Reno    NV   89509.0  United States   
215  New York    NY   12345.0  United States   
229  New York    NY   12345.0  United States   
237  New York    NY   12345.0  United States   
242       NaN   NaN       NaN            NaN   
244  New York    NY   12345.0  United States   
251  New York    NY   12345.0  United States   
277  New York    NY   12345.0  United States   

                                       contact  birthdate  weight  height  \
140     217-421-0935JohnTeichelmann@dayrep.com  12/3/1958   214.1      72   
153  JohnACarreiro@superrito.com1 775 770 7827   4/8/1976   177.8      68   
215                johndoe@email.com1234567890   1/1/1975   180.0      72   
229                johndoe@email.com1234567890   1/1/1975   180.0      72   
237                johndoe@email.com1234567890   1/1/1975   180.0      72   
242                                        NaN  2/25/1957   205.3      74   
244                johndoe@email.com1234567890   1/1/1975   180.0      72   
251                johndoe@email.com1234567890   1/1/1975   180.0      72   
277                johndoe@email.com1234567890   1/1/1975   180.0      72   

      bmi  
140  29.0  
153  27.0  
215  24.4  
229  24.4  
237  24.4  
242  26.4  
244  24.4  
251  24.4  
277  24.4

In [57]:
patient_df.address.value_counts()

123 Main Street             6
2778 North Avenue           2
2476 Fulton Street          2
648 Old Dear Lane           2
3094 Oral Lake Road         1
                           ..
1066 Goosetown Drive        1
4291 Patton Lane            1
4643 Reeves Street          1
174 Lost Creek Road         1
3652 Boone Crockett Lane    1
Name: address, Length: 483, dtype: int64

In [62]:
patient_df[patient_df.address == "2778 North Avenue"]

patient_id assigned_sex given_name  surname            address  city  \
97           98         male    Patrick  Gersten  2778 North Avenue  Burr   
502         503         male        Pat  Gersten  2778 North Avenue  Burr   

        state  zip_code        country                               contact  \
97         NE   68324.0  United States  PatrickGersten@rhyta.com402-848-4923   
502  Nebraska   68324.0  United States  PatrickGersten@rhyta.com402-848-4923   

    birthdate  weight  height   bmi  
97   5/3/1954   138.2      71  19.3  
502  5/3/1954   138.2      71  19.3

In [63]:
patient_df[patient_df.address == "2476 Fulton Street"]

patient_id assigned_sex given_name surname             address      city  \
131         132       female     Sandra  Taylor  2476 Fulton Street  Rainelle   
282         283       female      Sandy  Taylor  2476 Fulton Street  Rainelle   

    state  zip_code        country                               contact  \
131    WV   25962.0  United States  304-438-2648SandraCTaylor@dayrep.com   
282    WV   25962.0  United States  304-438-2648SandraCTaylor@dayrep.com   

      birthdate  weight  height   bmi  
131  10/23/1960   206.1      64  35.4  
282  10/23/1960   206.1      64  35.4

In [64]:
patient_df[patient_df.address == "648 Old Dear Lane"]

patient_id assigned_sex given_name   surname            address  \
24          25         male      Jakob  Jakobsen  648 Old Dear Lane   
29          30         male       Jake  Jakobsen  648 Old Dear Lane   

           city     state  zip_code        country  \
24  Port Jervis  New York   12771.0  United States   
29  Port Jervis  New York   12771.0  United States   

                                       contact birthdate  weight  height   bmi  
24  JakobCJakobsen@einrot.com+1 (845) 858-7707  8/1/1985   155.8      67  24.4  
29  JakobCJakobsen@einrot.com+1 (845) 858-7707  8/1/1985   155.8      67  24.4

In [65]:
patient_df

patient_id assigned_sex given_name      surname  \
0             1       female        Zoe      Wellish   
1             2       female     Pamela         Hill   
2             3         male        Jae       Debord   
3             4         male       Liêm         Phan   
4             5         male        Tim      Neudorf   
..          ...          ...        ...          ...   
498         499         male    Mustafa    Lindström   
499         500         male      Ruman      Bisliev   
500         501       female      Jinke    de Keizer   
501         502       female    Chidalu  Onyekaozulu   
502         503         male        Pat      Gersten   

                       address               city       state  zip_code  \
0         576 Brown Bear Drive  Rancho California  California   92390.0   
1    2370 University Hill Road          Armstrong    Illinois   61812.0   
2        1493 Poling Farm Road               York    Nebraska   68467.0   
3          2335 Webster Street         Woodbridge          NJ    7095.0   
4         1428 Turkey Pen Lane             Dothan          AL   36303.0   
..                         ...                ...         ...       ...   
498        2530 Victoria Court       Milton Mills          ME    3852.0   
499   494 Clarksburg Park Road             Sedona          AZ   86341.0   
500          649 Nutter Street      Overland Park          MO   64110.0   
501   3652 Boone Crockett Lane            Seattle          WA   98109.0   
502          2778 North Avenue               Burr    Nebraska   68324.0   

           country                                          contact  \
0    United States             951-719-9170ZoeWellish@superrito.com   
1    United States            PamelaSHill@cuvox.de+1 (217) 569-3204   
2    United States                 402-363-6804JaeMDebord@gustr.com   
3    United States       PhanBaLiem@jourrapide.com+1 (732) 636-8246   
4    United States                  334-515-7487TimNeudorf@cuvox.de   
..             ...                                              ...   
498  United States      207-477-0579MustafaLindstrom@jourrapide.com   
499  United States               928-284-4492RumanBisliev@gustr.com   
500  United States            816-223-6007JinkedeKeizer@teleworm.us   
501  United States  ChidaluOnyekaozulu@jourrapide.com1 360 443 2060   
502  United States             PatrickGersten@rhyta.com402-848-4923   

     birthdate  weight  height   bmi  
0    7/10/1976   121.7      66  19.6  
1     4/3/1967   118.8      66  19.2  
2    2/19/1980   177.8      71  24.8  
3    7/26/1951   220.9      70  31.7  
4    2/18/1928   192.3      27  26.1  
..         ...     ...     ...   ...  
498  4/10/1959   181.1      72  24.6  
499  3/26/1948   239.6      70  34.4  
500  1/13/1971   171.2      67  26.8  
501  2/13/1952   176.9      67  27.7  
502   5/3/1954   138.2      71  19.3  

[503 rows x 14 columns]

### Observation

#### Patient_df

* missed patients demographic information
* 503 entries
* Assigned_sex datatype is object
* Zip_code data type is float
* Birthdate data type is object
* Zip_code is not five digits
* contact: phone number and email not separated
* given_name and surname capitalized 
* Multiple entries for John Doe
* Patrick Gersten and Pat Gersten refers to the same person
* Sandra Taylor and Sandy Taylor refers to the same person
* Jakob Jakobsen and Jake Jakobsen refers to the same person

#### Adverse_df
* 34 entries
* given_name and surname in lowercase 
